Docs: armv7, pi 2, crosscompile (#3774)
mlpack/mlpack: doc/index.md

| ***Command-line programs*** | -- | [quickstart](quickstart/cli.md) | -- | [reference](user/bindings/cli.md) |
| ***Go*** | -- | [quickstart](quickstart/go.md) | -- | [reference](user/bindings/go.md) |

## mlpack on embedded systems

> mlpack is well suited for embedded systems due to the fact that it is written

Reviewer: ```suggestion
mlpack is well-suited for embedded systems due to the fact that it is written
```

> in C++ and it is header-only with minimal dependencies. In the following, we are
> adding a set of tutorials to allow you to experiment mlpack on various types of
> these systems.

Reviewer: ```suggestion
in C++ and it is header-only with minimal dependencies. Below are a set of
tutorials showing how mlpack can be used on various embedded systems.
```

* [cross-compile mlpack binding on armv7](embedded/crosscompile_armv7.md): a well
  guided tutorial that shows how to cross compile and run k-NN on

>   Raspberry PI 2.

Reviewer: Should we use `Raspberry Pi 2` directly in the same of the tutorial?  I guess that the tutorial is written specifically for that device, but I suppose the instructions will also work for other armv7 devices too?  (Maybe it would need a `BOARD_NAME` change?)

Author: we need to keep it arm7 because this is the architecture, the PI 2 is just a board implementing this architecture, the reason for that is to indicate to the user an example on the PI, but it is the same process whatever the board they are going to use.

Reviewer: I see, do you want to add some text at the top indicating that our example here is just for the RPi2, but any arm7 board will work?
